Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -1765,9 +1795,43 @@ void ACesium3DTileset::showTilesToRender(
 
     if (!Gltf->IsVisible()) {
       Gltf->SetVisibility(true, true);
-      Gltf->SetCollisionEnabled(ECollisionEnabled::QueryAndPhysics);
     }
+
+    Gltf->SetCollisionEnabled(ECollisionEnabled::QueryAndPhysics);
+  }
+}
+
+static void updateTileFade(Cesium3DTilesSelection::Tile* pTile) {
+  if (!pTile || !pTile->getContent().isRenderContent()) {
+    return;
+  }
+
+  if (pTile->getState() != Cesium3DTilesSelection::TileLoadState::Done) {
+    return;
+  }
+
+  const Cesium3DTilesSelection::TileContent& content = pTile->getContent();
+  const Cesium3DTilesSelection::TileRenderContent* pRenderContent =
+      content.getRenderContent();
+  if (!pRenderContent) {
+    return;
+  }
+
+  UCesiumGltfComponent* pGltf = reinterpret_cast<UCesiumGltfComponent*>(
+      pRenderContent->getRenderResources());
+  if (!pGltf) {
+    return;
   }
+
+  // Remap the fade percentage so that [0,0.5] --> [0,1] and [0.5,1] --> [1].
+  // This forces fading out tiles to stay opaque until the new tiles have fully
+  // faded in. This is needed for dithered fading, otherwise you will be able
+  // to partially see straight through both tiles during a transition.
```

Comment:
I don't know how practical this is, because I don't know how the dither function actually works, but I wanted to jot down this idea somewhere...

Here's how I think of the dither function...  At 0.0, the dither has no pixels active. At 1.0, the dither has all the pixels active. In between, the number of pixels active increases monotonically. A greater dither value will never deactivate any pixels that were previously active, it will only add new active pixels. All that is probably true, I'm pretty sure (though I might have the 0.0 and 1.0 backwards).

Now let's add one more property that might not be true, but maybe it is, and if it's not then I don't see why it couldn't be made true, at least...

The sequence of pixels activated is fully deterministic. At a value of 0.5, a pseudo-random set of pixels will be active, but it's always the _same_ pseudo-random set. I believe the dither is in screen-space, so when I refer to sets of pixels I mean on the screen.

If all of these things are true, then there's no need to fully fade in before fading out, we can do both simultaneously. For the tile fading in, we use a dither value from 0.0 to 1.0. For the fading out, we _also_ use a dither value from 0.0 to 1.0, but then we invert the opacity mask value, so we only show the tile where the dither is _not_ active. As long as both the fading in and the fading out tile use the same dither value, one tile will be shown exactly in the screen-space pixels where the other is not, at all times.

Entirely likely I've missed something, but that seems like it would work and give a nice visual quality improvement. No depth fighting, for one thing! Faster and smoother transitions for another. What do you think?